packages/amplify-cli/src/lib/pull-backend.js

```diff
@@ -0,0 +1,38 @@
+const { initializeEnv } = require('./initialize-env');
+const { constructInputParams, postPullCodeGenCheck } = require('./amplify-service-helper');
+
+async function pullBackend(context) {
+  context.exeInfo = context.amplify.getProjectDetails();
+  context.exeInfo.inputParams = constructInputParams(context);
+  context.print.info('');
+  context.print.info('Pre pull check:');
+  const hasChanges = await context.amplify.showResourceTable();
+  context.print.info('');
+
+  context.exeInfo.forcePush = false;
+  context.exeInfo.restoreBackend = !context.exeInfo.inputParams.amplify.noOverride;
+
+  if (hasChanges && context.exeInfo.restoreBackend) {
+    context.print.warning('Local changes detected.');
+    context.print.warning('Amplify pull will override your local changes with what is currently in the cloud.');
+    if (!context.exeInfo.inputParams.yes) {
+      context.print.info('It is recommended to perform a merge through codebase repository before you continue.');
+      const confirmOverride = await context.amplify.confirmPrompt.run('Do you want to continue?', false);
+      if (!confirmOverride) {
+        process.exit(0);
+      }
+    }
+  }
+
+  await initializeEnv(context);
+  await postPullCodeGenCheck(context);
```

Comment:
this should also be an event

Reply:
Agree.
But we've decided to leave it as an empty operation for now.

Reply:
Even so it should be fired as an event which can either be consumed or not. We should start making use of the event system we have built as it would open for other plugin  to consume these events

Reply:
I can mark this as todo, currently there is no pre-pull and post-pull events yet.